src/fides/api/ctl/routes/system.py

```diff
@@ -0,0 +1,211 @@
+from typing import List
+
+from fastapi import Depends, HTTPException, Security
+from fastapi_pagination import Page, Params
+from fastapi_pagination.bases import AbstractPage
+from fastapi_pagination.ext.sqlalchemy import paginate
+from loguru import logger
+from pydantic.types import conlist
+from sqlalchemy.orm import Session
+from starlette.status import (
+    HTTP_200_OK,
+    HTTP_400_BAD_REQUEST,
+    HTTP_404_NOT_FOUND,
+    HTTP_422_UNPROCESSABLE_ENTITY,
+)
+
+from fides.api.ctl.sql_models import System  # type: ignore[attr-defined]
+from fides.api.ctl.utils.api_router import APIRouter
+from fides.api.ops.api import deps
+from fides.api.ops.api.v1.endpoints.connection_endpoints import (
+    requeue_requires_input_requests,
+    validate_secrets,
+)
+from fides.api.ops.api.v1.scope_registry import (
+    CONNECTION_CREATE_OR_UPDATE,
+    CONNECTION_READ,
+)
+from fides.api.ops.api.v1.urn_registry import (
+    CONNECTION_TYPES,
+    SYSTEM_CONNECTIONS,
+    V1_URL_PREFIX,
+)
+from fides.api.ops.models.connectionconfig import ConnectionConfig
+from fides.api.ops.schemas.api import BulkUpdateFailed
+from fides.api.ops.schemas.connection_configuration.connection_config import (
+    BulkPutConnectionConfiguration,
+    ConnectionConfigurationResponse,
+    CreateConnectionConfigurationWithSecrets,
+    SaasConnectionTemplateValues,
+)
+from fides.api.ops.service.connectors.saas.connector_registry_service import (
+    create_connection_config_from_template_no_save,
+    load_registry,
+    registry_file,
+)
+from fides.api.ops.util.oauth_util import verify_oauth_client
+from fides.lib.exceptions import KeyOrNameAlreadyExists
+
+router = APIRouter(tags=["System"], prefix=f"{V1_URL_PREFIX}{SYSTEM_CONNECTIONS}")
+
+
+def validate_system(db: Session, fides_key: str) -> System:
+    system = System.get_by(db, field="fides_key", value=fides_key)
+    if system is None:
+        raise HTTPException(
+            status_code=HTTP_404_NOT_FOUND,
+            detail="A valid system must be provided to create or update connections",
+        )
+    return system
+
+
+@router.get(
+    "",
+    dependencies=[Security(verify_oauth_client, scopes=[CONNECTION_READ])],
```

Comment:
It's an interesting side effect of combining ctl and ops-like routes, but the base system routes are not authenticated but you need a token with scopes to get the system connections.  I think Thomas is working on adding authentication to some of the other ctl routes.

Reply:
There are some interesting differences like sync vs async db sessions.

Reply:
oh gosh yes that distinction between ctl and ops endpoints is causing headaches